### PR TITLE
Robert/testing sdk

### PR DESCRIPTION
**Pull request by robert-zaremba**
## Description

* added helper functions to create a test Cosmos SDK Context and KVStore
* demonstrated the usage in:
  * converted uibc quota test to an unit test
  * missing test for the `Iterate` function.

### CODE REVIEW

File: tests/tsdk/store.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+	"github.com/cosmos/cosmos-sdk/store"
+	types "github.com/cosmos/cosmos-sdk/store/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/stretchr/testify/assert"
+	"github.com/tendermint/tendermint/libs/log"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+	dbm "github.com/tendermint/tm-db"
+)
+
+// NewCommitMultiStore creats SDK Multistore
```
Comment (toteki) [inline tests/tsdk/store.go:16]: Is there a way to document (or link to docs) the purpose of tkeys and keys for those unfamiliar with the SDK?

Comment (robert-zaremba) [reply tests/tsdk/store.go:16]: the type already provides that. Also, we are having it in every module we are implementing.

File: util/store/iter_test.go [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+		{[]byte{1, 8}, []byte{2}},
+		{[]byte{2, 4}, []byte{1}},
+	}
+
+	db := tsdk.NewKVStore(t)
+	db.Set(pairs[0].K, pairs[0].V)
+	db.Set(pairs[4].K, pairs[4].V)
+	db.Set(pairs[3].K, pairs[3].V)
+	db.Set(pairs[1].K, pairs[1].V)
+	db.Set(pairs[2].K, pairs[2].V)
```
Comment (toteki) [inline util/store/iter_test.go:27]: for / range loop might make sense here, or a comment indicating why it won't (e.g. `pairs` is sorted but we want to `db.Set` the keys out of order to properly test iterator)

Comment (robert-zaremba) [reply util/store/iter_test.go:27]: BTW, would be great if you add more edge cases and also add a test to the reverse iterator. I'm also planning to write a generic fetcher, but that will be a new feature.

**Review by gsk967 — APPROVED** [review]
LGTM

File: util/store/iter_test.go (hunk 2 above)
Comment (robert-zaremba) [reply util/store/iter_test.go:30]: @toteki I will create a task to add missing tests. This PR is meant to provide the helper functions I mentioned with example how to use it.

**Review by toteki — APPROVED** [review]
